adafruit_pioasm.py

```diff
@@ -79,6 +136,89 @@ def __init__(self, text_program: str, *, build_debuginfo: bool = False) -> None:
             elif line.startswith(".side_set"):
                 sideset_count = int(line.split()[1], 0)
                 sideset_enable = "opt" in line
+            elif line.startswith(".fifo"):
+                require_before_instruction()
+                fifo_type = line.split()[1]
+                required_version = FIFO_TYPES.get(fifo_type)
+                if required_version is None:
+                    raise RuntimeError(f"Invalid fifo type {fifo_type}")
+                require_version(required_version, line)
+            elif line.startswith(".mov_status"):
+                require_before_instruction()
+                required_version = 0
+                mov_status_n = 0
+                mov_status_type = words[1]
+                if words[1] in ("txfifo", "rxfifo"):
+                    if words[2] != "<":
+                        raise RuntimeError(f"Invalid {line}")
+                    mov_status_n = int_in_range(words[3], 0, 32, words[1])
+                elif words[1] == "irq":
+                    required_version = 1
+                    idx = 2
+                    if words[idx] == "next":
+                        mov_status_n = 0x10
+                        idx += 1
+                    elif words[idx] == "prev":
+                        mov_status_n = 0x8
+                        idx += 1
+                    else:
+                        mov_status_n = 0
+                    if words[idx] != "set":
+                        raise RuntimeError(f"Invalid {line})")
+                    mov_status_n |= int_in_range(words[idx + 1], 0, 8, "mov_status irq")
+                require_version(required_version, line)
+            elif words[0] == ".out":
+                require_before_instruction()
+                out_count = int_in_range(
+                    words[1], 32 if pio_version == 0 else 1, 33, ".out count"
```

Comment:
Does this mean 32 is the only out_count valid for version 0? It looks like this is a copy/paste issue from `.in`.

Reply:
yup I think it was incorrect.